File: 02 - Python/helloworld.py
# Hola mundo

print("Hola, Python")

'''
Este
es
un
comentario
'''

my_string = "Esto es una cadena de texto"
my_string = "Aquí cambio el valor de la cadena de texto"
print(type(my_string))
print(my_string)

my_string = 6 # Tipado dinámico
print(type(my_string))
print(my_string)

my_other_string: str = "Esto es otra cadena de texto"
# my_other_string = 10 # Se cambiaría el tipo igualmente

my_int = 7
my_int = my_int + 4
print(my_int)
print(my_int - 1)
print(my_int)

my_float = 6.5
print(type(my_float))
print(my_float)

print(my_int + my_float)
# print(my_int + my_float + my_other_string) # Error

my_bool = True
my_bool = False
print(type(my_bool))
print(my_bool)

print("El valor de mi entero es " + str(my_int) + " y el de mi bool " + str(my_bool))
print(f"El valor de mi entero es {my_int} y el de mi bool {my_bool}")

my_list = [my_other_string, my_int, my_float, my_bool]
print(type(my_list))
my_list.append(my_bool)
print(my_list)
print(my_list[0])
# print(my_list[4]) # Error

my_dict = {"string": my_other_string, "int": my_int, "float": my_float, "bool": my_bool, "mouredev": "Brais Moure"}
print(type(my_dict))
print(my_dict)
print(my_dict["mouredev"])

my_set = {my_other_string, my_int, my_float, my_bool, my_bool, my_bool}
print(type(my_set))
print(my_set)

my_tuple = (my_other_string, my_int, my_float, my_bool, my_bool, my_bool)
print(type(my_tuple))
print(my_tuple)

if my_int == 11 or my_bool == True:
    print("El valor es 11")
elif my_int == 12:
    print("El valor es 12")
else:
    print("El valor no es 11 ni 12")

for my_item in my_list:
    print(my_item)

for my_item in range(10):
    print(my_item)

for my_item in my_dict:
    print(my_item)

for my_item in my_set:
    print(my_item)

for my_item in my_tuple:
    print(my_item)

def my_function():
    print("Esto es una funcion")

for my_item in range(10):
    my_function()

def my_function_with_return(param):
    return 10 + param

my_return = my_function_with_return(5)
print(my_return)

class MyClass:

    def __init__(self, my_name):
        self.my_name = my_name

    def print_name(self):
        print(self.my_name)


my_class = MyClass("Brais")
my_class.print_name()
my_class.my_name = "MoureDev"
print(type(my_class))
print(my_class.my_name)

File: 11 - CSharp/HelloWorld.cs
// Día 11: C#
// Clase en vídeo: https://youtu.be/L-f8u0hwi4Y

using System;

namespace CSharpHelloWorld
{
    class HelloWorld
    {
        static void Main(string[] args)
        {
            // Hola mundo
            Console.WriteLine("Hola, C#");

            /*
            Esto
            es
            un
            comentario
            */

            // Variables

            string myString = "Esto es una cadena de texto";
            myString = "Aquí cambio el valor de la cadena de texto";
            Console.WriteLine(myString);

            int myInt = 7;
            myInt = myInt + 4;
            Console.WriteLine(myInt);
            Console.WriteLine(myInt - 1);
            Console.WriteLine(myInt);

            double myDouble = 6.5;
            Console.WriteLine(myDouble);

            float myFloat = 6.5f;
            Console.WriteLine(myFloat);

            Console.WriteLine(myInt + myDouble + myFloat);

            bool myBool = true;
            myBool = false;
            Console.WriteLine(myBool);

            dynamic myDynamic = 6;
            myDynamic = "Mi dato dinámico";
            Console.WriteLine(myDynamic + myFloat);

            var myVar = "Mi variable de tipo inferido";
            // myVar = 6; Error
            Console.WriteLine(myVar);

            Console.WriteLine($"El valor de mi entero es {myInt} y el de mi bool {myBool}");

            // Constantes

            const string MyConst = "Mi constante";
            Console.WriteLine(MyConst);

            // Arrays

            var myArray = new string[] { "Brais", "Moure", "MoureDev" };
            Console.WriteLine(myArray[0]);

            myArray[2] = "36";
            Console.WriteLine(myArray[2]);

            // Diccionarios

            var myDictionary = new Dictionary<string, int>
            {
                {"mouredev", 36},
                {"brookslynx", 5},
                {"cabreragreta", 25}
            };

            // Sets

            Console.WriteLine(myDictionary["mouredev"]);

            var mySet = new HashSet<string> { "Brais", "Moure", "MoureDev", "MoureDev" };

            // Tuplas

            var myTuple = ("Brais", "Moure", "MoureDev");
            Console.WriteLine(myTuple);

            // Bucles

            for (int index = 0; index < 10; index++)
            {
                Console.WriteLine(index);
            }

            foreach (var myItem in myArray)
            {
                Console.WriteLine(myItem);
            }

            foreach (var myItem in myDictionary)
            {
                Console.WriteLine(myItem);
            }

            foreach (var myItem in mySet)
            {
                Console.WriteLine(myItem);
            }

            // Control de flujo

            if (myInt == 11 && myBool == true)
            {
                Console.WriteLine("El valor es 11");
            }
            else if (myInt == 12 || myBool == false)
            {
                Console.WriteLine("El valor es 12");
            }
            else
            {
                Console.WriteLine("El valor no es 11 ni 12");
            }

            // Funciones

            MyFunction();
            Console.WriteLine(MyFunctionWithReturn(5));

            // Clases

            var myClass = new MyClass("Brais");
            myClass.myName = "MoureDev";
            Console.WriteLine(myClass.myName);
        }
        static void MyFunction()
        {
            Console.WriteLine("Mi función");
        }

        static int MyFunctionWithReturn(int param)
        {
            return 10 + param;
        }

        class MyClass
        {
            public string myName { get; set; }

            public MyClass(string myName)
            {
                this.myName = myName;
            }
        }
    }


}